Re-scss absensi component

aamesaa opened the pull request:
Rename absesnsi component and add data binding. A little change on `NavBar`

aamesaa committed:
combine upload and employee style

aamesaa committed:
remove style.scss

aamesaa committed:
Fix login background

aamesaa committed:
Fix Shift and newShift page scss

Rename style class and add object on Shift page

aamesaa committed:
Fix Employee page

- remove search employee component and move to Employee
- move the thead from TableEmp to Employee
- Use flex to display search employee
- add object to show employee
- the page is already responsive

aamesaa committed:
rename newEmployee styling class

rename newEmployee styling class

aamesaa committed:
Presensi styling

- Move search area into Presensi.vue
- Move table header to Presensi.vue
- Add modal component for editingTapHour

aamesaa committed:
remove container on app.vue

aamesaa committed:
minor adjustment

aamesaa committed:
Merge branch 'vue' into re-scss-absensi

riencus commented on src/component/LoginForm.vue:
```diff
@@ -42,11 +42,12 @@ export default {
 
 <style lang="scss" scoped="true">
-  main {
+  .login {
```
I think you forgot to use the class, CMIIW I don't see any `class="login"` anywhere

aamesaa replied:
Ok. I think I've missed that.

riencus commented on src/component/absensi/Employee.vue:
```diff
@@ -1,46 +1,165 @@
+                          :item="item">
+               </table-emp>
+            </tbody>
+        </table>
```
This is minor, but I think the mis-indentation on `</table>` is annoying

riencus commented on src/component/absensi/Upload.vue:
```diff
@@ -1,47 +1,70 @@
+            padding: 7px 15px;
+            font-family:Arial;
+            font-size:inherit;
+            &[type=file]{
```
Why use this selector? Since I don't see any other tag using `upload__input` class, why don't we insert `border: none` into the `upload__input` class?

riencus commented on src/component/absensi/employee/TableEmp.vue:
```diff
@@ -1,67 +1,27 @@
+        <td>{{item.chiefName}}</td>
+        <td>{{item.startWorkingDate}}</td>
+    </tr>
 </template>
```
If this class only contains a `<tr>` tag, why don't we just delete this class and use `v-for` on the `Employee.vue` directly?
```
<tr v-for="employee in employees">
        <td>{{employee.nik}}</td>
        <td>{employee.fullname}}</td>
        ...
</tr>
```
Or... if We want to make it separate, how about make the whole table to be included inside this file?
```
<template>
    <table>
    ....
    </table>
</template>
````

aamesaa replied:
At first, the whole table was separated, but after I use `v-for` it also loop the `<thead>`. So, i separated the component inside of `<tbody>`. Should we directly use `v-for` or separate the table component?

riencus replied:
Let's discuss it. First, I want to know why did you separate the table in the first place. Why did you separate them?

aamesaa replied:
my bad. I just realize that i've directly use `v-for` in `Shift.vue`. Gonna change it soon

riencus commented on src/component/absensi/shift/NewShift.vue:
```diff
@@ -1,22 +1,22 @@
-                <label for="department"> Department</label>:
-                <select name="department" id="department" >
+            <div class="newShift__form__section">
+                <label for="department" class="newShift__form__section__label"> Department</label>
```
Bem naming [doesn't recommend](http://getbem.com/faq/#css-nested-elements) Us to use `block__element__element...` We should change the class name.
How about `new-shift__label`?

aamesaa replied:
I changed it into `new-shift__label`, `new-shift__input` and `new-shift__section`

riencus commented on src/component/absensi/shift/Shift.vue:
```diff
@@ -19,74 +17,93 @@
-
+                <tr v-for="item in shiftItems"
+                    :key="item.id"
+                    :item="item">
```
These two attributes (`key` and `item`) are unnecessary; only to be used when `v-for`  is [used on a component](https://vuejs.org/v2/guide/list.html#Components-and-v-for)

aamesaa replied:
in the shift page, we need to get the nik for updating the TapIn / TapOut. I think I missplace and using the wrong syntax. I have fix it in the last commit

riencus replied:
@aamesaa the `:key="item.id" :item="item">`is still there, does it breaks the whole system when those 2 is not removed?

aamesaa replied:
My bad. I thought this was Presensi page, so I said that we need `:key="item.id" :item="item"` and made a change on `Presensi.vue`. I've been fixed this on the latest commit

riencus commented on src/component/absensi/shift/Shift.vue:
```diff
@@ -19,74 +17,93 @@
+            tbody{
+                tr:hover{
+                    background-color: #e1e1e1
+                }
```
How about `shift__column:hover { ... } `? It's more BEM friendly and has more direct meaning in my opinion.

aamesaa replied:
I removed the `table-hover`, and add `table__body` to put the `tr:hover`

riencus replied:
why not make a class just for `<tr>` instead of giving `:hover` into the `<tr>` tag?

riencus reviewed (changes requested):
There's some class naming problem and the usage of vue component that needs some discussion first.

aamesaa committed:
Merge vue into re-scss-absensi

aamesaa committed:
Change based on review in PR #10

- Add login class on LoginForm.vue
- Remove the &[type=file] style
- Rename class inside the form
- Reindent tag inside the employee

aamesaa committed:
Remove table-hover class

remove table-hover class and add table__body instead

aamesaa committed:
Move the table content in one file

aamesaa committed:
Add selected shift array

aamesaa commented:
All the requested change have been committed

aamesaa committed:
Remove :item and :key in Shift
